net/veth.go

```diff
@@ -186,6 +186,15 @@ func AttachContainer(netNSPath, id, ifName, bridgeName string, mtu int, withMult
 		return err
 	}
 
+	// set hairpin mode
+	veth, err := netlink.LinkByName(name)
```

Comment:
Did you mean over [here](https://github.com/deitch/weave/blob/b75543d0e5ab8733fb9af4494742445ab8943c50/net/veth.go#L127)?

I thought about it, but there were 2 reasons I didn't.

1. That only gets called f the interface does not yet exist in the namespace [here](https://github.com/deitch/weave/blob/b75543d0e5ab8733fb9af4494742445ab8943c50/net/veth.go#L121). In theory that should never happen, but...
2. Isn't the argument to the callback the side of the veth *in* the container? Don't we want to set hairpin on the side attached to the bridge?

For the second, I guess we could either (a) pass both back in the callback or (b) determine it from the veth.

Out of curiosity, why is it in a callback and not in the next step? Is it just to wrap cleanup?